presto-main/src/test/java/com/facebook/presto/cost/TestScalarStatsCalculator.java

```diff
@@ -316,6 +316,39 @@ public void testNonDivideArithmeticBinaryExpression()
                 .averageRowSize(2.0);
     }
 
+    @Test
+    public void testArithmeticBinaryWithZero()
+    {
+        PlanNodeStatsEstimate relationStats = PlanNodeStatsEstimate.builder()
+                .addSymbolStatistics(new Symbol("x"), SymbolStatsEstimate.builder()
+                        .setLowValue(-1)
+                        .setHighValue(10)
+                        .setDistinctValuesCount(4)
+                        .setNullsFraction(0.1)
+                        .setAverageRowSize(0)
+                        .build())
+                .addSymbolStatistics(new Symbol("zero"), SymbolStatsEstimate.zero())
+                .setOutputRowCount(10)
+                .build();
+
+        assertCalculate(expression("x + zero"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
```

Comment:
why not `x` stats?

Reply:
Let me do the renames. `zero -> all_nulls`. `all_nulls` column has `SymbolStatsEstimate.zero()` symbol stats (0 NDV, empty range, etc.)